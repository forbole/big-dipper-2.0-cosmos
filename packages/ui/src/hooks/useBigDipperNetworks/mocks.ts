import { ChainIdDocument } from '@/graphql/types/general_types';
import { query } from '@/hooks/useBigDipperNetworks';

const mockQueryResult = jest.fn().mockReturnValue({
  data: {
    networks: [
      {
        name: 'Desmos',
        logo: 'https://raw.githubusercontent.com/forbole/big-dipper-networks/main/logos/desmos.svg?sanitize=true',
        cover:
          'https://raw.githubusercontent.com/forbole/big-dipper-networks/main/covers/desmos.png?sanitize=true',
        links: [
          {
            name: 'Testnet',
            chain_id: 'morpheus-apollo-3',
            url: 'https://morpheus.desmos.network',
          },
          {
            name: 'Mainnet',
            chain_id: 'desmos-mainnet',
            url: 'https://explorer.desmos.network',
          },
        ],
      },
    ],
  },
});

export const mockQuery = { request: { query }, result: mockQueryResult };

const mockChainIdResult = jest.fn().mockReturnValue({
  data: {
    genesis: [
      {
        chainId: 'desmos-mainnet',
        time: '',
      },
    ],
  },
});

export const mockChainId = { request: { query: ChainIdDocument }, result: mockChainIdResult };
