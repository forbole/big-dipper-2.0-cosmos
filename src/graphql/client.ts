import WebSocket from 'isomorphic-ws';
import {
  ApolloClient,
  InMemoryCache,
  split,
  HttpLink,
  ApolloLink,
  concat,
} from '@apollo/client';
import {
  getMainDefinition,
} from '@apollo/client/utilities';
import { WebSocketLink } from '@apollo/client/link/ws';

import { useMemo } from 'react';

const defaultOptions:any = {
  watchQuery: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
  query: {
    fetchPolicy: 'no-cache',
    errorPolicy: 'all',
  },
};

let apolloClient;

const httpLink = new HttpLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_URL,
});

const wsLink = new WebSocketLink({
  uri: process.env.NEXT_PUBLIC_GRAPHQL_WS ?? 'wss://localhost:3000',
  options: {
    reconnect: true,
  },
  webSocketImpl: WebSocket,
});

const link = typeof window !== 'undefined' ? split(
  ({ query }) => {
    const {
      kind, operation,
    }:any = getMainDefinition(query);
    return kind === 'OperationDefinition' && operation === 'subscription';
  },
  wsLink,
  httpLink,
) : httpLink;

const authMiddleware = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: {
    },
  });

  return forward(operation);
});

function createApolloClient() {
  const client = new ApolloClient({
    ssrMode: typeof window === 'undefined',
    link: concat(authMiddleware, link),
    cache: new InMemoryCache({
    }),
  });

  client.defaultOptions = defaultOptions;

  return client;
}

export function initializeApollo(initialState = null) {
  // eslint-disable-next-line
  const _apolloClient = apolloClient ?? createApolloClient();

  // If your page has Next.js data fetching methods that use Apollo Client, the initial state
  // gets hydrated here
  if (initialState) {
    // Get existing cache, loaded during client side data fetching
    const existingCache = _apolloClient.extract();
    // Restore the cache using the data passed from getStaticProps/getServerSideProps
    // combined with the existing cached data
    _apolloClient.cache.restore({
      ...existingCache, ...initialState,
    });
  }
  // For SSG and SSR always create a new Apollo Client
  if (typeof window === 'undefined') return _apolloClient;
  // Create the Apollo Client once in the client
  if (!apolloClient) apolloClient = _apolloClient;

  return _apolloClient;
}

export function useApollo(initialState) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
