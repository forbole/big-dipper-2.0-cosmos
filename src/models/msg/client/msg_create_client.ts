import { Categories } from '../types';

class MsgCreateClient {
    public category: Categories;
    public type: string;
    public signer: string;
    public chainId: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.chainId = payload.chainId;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgCreateClient({
        json,
        type: json['@type'],
        signer: json.signer,
        chainId: json.client_state?.chain_id,
      });
    }
}

export default MsgCreateClient;
