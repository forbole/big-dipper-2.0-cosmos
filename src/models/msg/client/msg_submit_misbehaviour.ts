import { Categories } from '../types';

class MsgSubmitMisbehaviour {
    public category: Categories;
    public type: string;
    public signer: string;
    public clientId: string;
    public misbehaviour: string;
    public json: any;

    constructor(payload: any) {
      this.category = 'client';
      this.type = payload.type;
      this.signer = payload.owner;
      this.clientId = payload.clientId;
      this.misbehaviour = payload.misbehaviour;
      this.json = payload.json;
    }

    static fromJson(json: any) {
      return new MsgSubmitMisbehaviour({
        json,
        type: json['@type'],
        signer: json.signer,
        clientId: json.client_id,
        misbehaviour: json.misbehaviour,
      });
    }
}

export default MsgSubmitMisbehaviour;
