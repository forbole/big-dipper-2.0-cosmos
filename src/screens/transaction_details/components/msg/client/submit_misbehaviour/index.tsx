import React from 'react';
import Trans from 'next-translate/Trans';
import { Typography } from '@material-ui/core';
import { Name } from '@components';
import { MsgSubmitMisbehaviour } from '@models';
import { useChainContext } from '@contexts';

const SubmitMisbehaviour = (props: {
    message: MsgSubmitMisbehaviour;
}) => {
  const { findAddress } = useChainContext();
  const { message } = props;

  const signer = findAddress(message.signer);
  const signerMoniker = signer ? signer?.moniker : message.signer;

  return (
    <Typography>
      <Trans
        i18nKey="message_contents:txSubmitMisbehaviourContent"
        components={[
          (
            <Name
              address={message.signer}
              name={signerMoniker}
            />
          ),
          <b />,
        ]}
        values={{
          clientId: message.clientId,
        }}
      />
    </Typography>
  );
};

export default SubmitMisbehaviour;
