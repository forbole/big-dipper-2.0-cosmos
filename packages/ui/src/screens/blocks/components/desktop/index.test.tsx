import Desktop from '@/screens/blocks/components/desktop';
import MockTheme from '@/tests/mocks/MockTheme';
import { mockClient } from '@/tests/mocks/mockApollo';
import { ApolloProvider } from '@apollo/client';
import { ComponentProps } from 'react';
import renderer from 'react-test-renderer';
import AutoSizer from 'react-virtualized-auto-sizer';

// ==================================
// mocks
// ==================================
jest.mock('@/components/loading', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Loading" {...props} />
));
jest.mock('@/components/avatar_name', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="AvatarName" {...props} />
));

jest.mock(
  'react-virtualized-auto-sizer',
  () =>
    ({ children }: ComponentProps<typeof AutoSizer>) =>
      children({
        height: 600,
        width: 600,
      })
);

// ==================================
// unit tests
// ==================================
describe('screen: Home/Blocks/Desktop', () => {
  it('matches snapshot', () => {
    const component = renderer.create(
      <ApolloProvider client={mockClient}>
        <MockTheme>
          <Desktop
            items={[
              {
                height: 300,
                txs: 2,
                timestamp: '',
                proposer: 'address',
                hash: 'hash',
              },
              {
                height: 301,
                txs: 2,
                timestamp: '',
                proposer: 'address',
                hash: 'hash',
              },
            ]}
            itemCount={2}
            loadMoreItems={() => jest.fn()}
            isItemLoaded={() => true}
          />
        </MockTheme>
      </ApolloProvider>
    );
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
