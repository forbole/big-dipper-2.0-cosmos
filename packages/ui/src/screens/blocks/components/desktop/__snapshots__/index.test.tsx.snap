// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Home/Blocks/Desktop matches snapshot 1`] = `
.emotion-0 {
  height: 100%;
}

.emotion-1 {
  background: #FFFFFF;
  height: 50px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1.odd {
  background: #F8F8F8;
}

.emotion-1>* {
  width: 100%;
}

.emotion-1 .MuiTypography-body1 {
  white-space: nowrap;
}

.emotion-2 {
  margin: 0;
  font-size: 1rem;
  letter-spacing: 0.15px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.235;
}

.emotion-8 {
  margin: 0;
  font-size: 1rem;
  letter-spacing: 0.15px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.235;
  text-align: right;
}

.emotion-11 {
  background: #FFFFFF;
  height: 50px;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #414141;
}

.emotion-11.odd {
  background: #F8F8F8;
}

.emotion-11>* {
  width: 100%;
}

.emotion-11 .MuiTypography-body1 {
  white-space: nowrap;
}

.emotion-12 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
}

.emotion-18 {
  margin: 0;
  font-size: 1rem;
  white-space: pre-wrap;
  letter-spacing: 0.5px;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  line-height: 1.5;
  text-align: right;
}

<div
  className="emotion-0"
>
  <div
    onScroll={[Function]}
    style={
      {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 50,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        {
          "height": 50,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    >
      <div
        className="emotion-1"
        style={
          {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 emotion-2"
        >
          height
        </h4>
      </div>
      <div
        className="emotion-1"
        style={
          {
            "height": 50,
            "left": 90,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 150,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 emotion-2"
        >
          proposer
        </h4>
      </div>
      <div
        className="emotion-1"
        style={
          {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 150,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 emotion-2"
        >
          hash
        </h4>
      </div>
      <div
        className="emotion-1"
        style={
          {
            "height": 50,
            "left": 390,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight emotion-8"
        >
          txs
        </h4>
      </div>
      <div
        className="emotion-1"
        style={
          {
            "height": 50,
            "left": 480,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <h4
          className="MuiTypography-root MuiTypography-h4 MuiTypography-alignRight emotion-8"
        >
          time
        </h4>
      </div>
    </div>
  </div>
  <div
    className="scrollbar"
    onScroll={[Function]}
    style={
      {
        "WebkitOverflowScrolling": "touch",
        "direction": "ltr",
        "height": 550,
        "overflow": "auto",
        "position": "relative",
        "width": 600,
        "willChange": "transform",
      }
    }
  >
    <div
      style={
        {
          "height": 100,
          "pointerEvents": undefined,
          "width": 600,
        }
      }
    >
      <div
        className="odd emotion-11"
        style={
          {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          <a
            className="value"
            href="/blocks/300"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            300
          </a>
        </div>
      </div>
      <div
        className="odd emotion-11"
        style={
          {
            "height": 50,
            "left": 90,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 150,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          <div
            address="address"
            id="AvatarName"
            imageUrl=""
            name="address"
          />
        </div>
      </div>
      <div
        className="odd emotion-11"
        style={
          {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 150,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          hash
        </div>
      </div>
      <div
        className="odd emotion-11"
        style={
          {
            "height": 50,
            "left": 390,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 90,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight emotion-18"
        >
          2
        </div>
      </div>
      <div
        className="odd emotion-11"
        style={
          {
            "height": 50,
            "left": 480,
            "position": "absolute",
            "right": undefined,
            "top": 0,
            "width": 120,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight emotion-18"
        />
      </div>
      <div
        className="emotion-11"
        style={
          {
            "height": 50,
            "left": 0,
            "position": "absolute",
            "right": undefined,
            "top": 50,
            "width": 90,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          <a
            className="value"
            href="/blocks/301"
            onClick={[Function]}
            onMouseEnter={[Function]}
            onTouchStart={[Function]}
          >
            301
          </a>
        </div>
      </div>
      <div
        className="emotion-11"
        style={
          {
            "height": 50,
            "left": 90,
            "position": "absolute",
            "right": undefined,
            "top": 50,
            "width": 150,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          <div
            address="address"
            id="AvatarName"
            imageUrl=""
            name="address"
          />
        </div>
      </div>
      <div
        className="emotion-11"
        style={
          {
            "height": 50,
            "left": 240,
            "position": "absolute",
            "right": undefined,
            "top": 50,
            "width": 150,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 emotion-12"
        >
          hash
        </div>
      </div>
      <div
        className="emotion-11"
        style={
          {
            "height": 50,
            "left": 390,
            "position": "absolute",
            "right": undefined,
            "top": 50,
            "width": 90,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight emotion-18"
        >
          2
        </div>
      </div>
      <div
        className="emotion-11"
        style={
          {
            "height": 50,
            "left": 480,
            "position": "absolute",
            "right": undefined,
            "top": 50,
            "width": 120,
          }
        }
      >
        <div
          className="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight emotion-18"
        />
      </div>
    </div>
  </div>
</div>
`;
